src/wp-includes/block-template-utils.php

```diff
@@ -1430,6 +1430,17 @@ function get_template_hierarchy( $slug, $is_custom = false, $template_prefix = '
 		$template_hierarchy[] = 'singular';
 	}
 	$template_hierarchy[] = 'index';
+
+	$template_type = '';
+	if ( ! empty( $template_prefix ) ) {
+		list( $template_type ) = explode( '-', $template_prefix );
+	} else {
+		list( $template_type ) = explode( '-', $slug );
+	}
+	$valid_template_types = array( '404', 'archive', 'attachment', 'author', 'category', 'date', 'embed', 'frontpage', 'home', 'index', 'page', 'paged', 'privacypolicy', 'search', 'single', 'singular', 'tag', 'taxonomy' );
```

Comment:
I guess you got these from: https://developer.wordpress.org/reference/hooks/type_template_hierarchy/, right?

The logic for checking whether to apply the filter or not, I guess could go on top of the function, since we already have some early returns.

Regarding the actual logic for template_type, I'd love some eyes from @jorgefilipecosta, since while I introduced this function, he has updated it afterwards.

Reply:
> I guess you got these from: https://developer.wordpress.org/reference/hooks/type_template_hierarchy/, right?

Yup, directly from the documented code, in fact:

https://github.com/WordPress/wordpress-develop/blob/6.4/src/wp-includes/template.php#L39-L56

> The logic for checking whether to apply the filter or not, I guess could go on top of the function, since we already have some early returns.

Good catch, I missed those early returns. In c83b3ac502f82396d743d4e07de18dce8304d8f6 I updated the code so we call `apply_filter()` on those early returns as well. It's not 100% clear to me if that's what you were proposing or you have a better idea?

Reply:
That was it, thank you! We just need a quick look from @jorgefilipecosta now :)